components/metrics/handler.go

```diff
@@ -0,0 +1,92 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+)
+
+var (
+	handlerLabelKeys = []string{
+		labelKeyHandlerName,
+		labelSuccess,
+	}
+
+	// handlerExecutionTimeBuckets are one order of magnitude smaller than default buckets (5ms~10s),
+	// because the handler execution times are typically shorter (µs~ms range).
+	handlerExecutionTimeBuckets = []float64{
+		0.0005,
+		0.001,
+		0.0025,
+		0.005,
+		0.01,
+		0.025,
+		0.05,
+		0.1,
+		0.25,
+		0.5,
+		1,
+	}
+)
+
+type HandlerPrometheusMetricsMiddleware struct {
+	handlerExecutionTimeSeconds *prometheus.HistogramVec
+	handlerCountTotal           prometheus.Gauge
+}
+
+func (m HandlerPrometheusMetricsMiddleware) Middleware(h message.HandlerFunc) message.HandlerFunc {
+
+	m.handlerCountTotal.Inc()
```

Comment:
not needed?

Reply:
seen it also below

Reply:
Removed all the `*CountTotal` metrics, as we found another way to count it in the Grafana dashboard itself e9f08ad5792e7dea950c368e1edd9e14b9e79226